tip-distributor/src/claim_mev_workflow.rs

```diff
@@ -68,318 +50,286 @@ pub enum ClaimMevError {
+        ("claim_statuses_onchain", claim_statuses.len(), i64),
+    );
+
+    let transactions = build_transactions_new(
+        tip_distribution_program_id,
+        merkle_trees,
+        tdas,
+        claimants,
+        claim_statuses,
+        micro_lamports,
+        payer_pubkey,
+    );
+
+    Ok(transactions)
+}
+
+pub async fn claim_mev_tips(
+    merkle_trees: &GeneratedMerkleTreeCollection,
+    rpc_url: String,
+    tip_distribution_program_id: Pubkey,
+    keypair: Arc<Keypair>,
+    max_loop_duration: Duration,
+    micro_lamports: u64,
+) -> Result<(), ClaimMevError> {
+    let rpc_client = RpcClient::new_with_timeout_and_commitment(
+        rpc_url,
+        Duration::from_secs(300),
+        CommitmentConfig::confirmed(),
+    );
+
+    let start = Instant::now();
+    while start.elapsed() <= max_loop_duration {
+        let mut all_claim_transactions = get_claim_transactions_for_valid_unclaimed(
+            &rpc_client,
+            merkle_trees,
             tip_distribution_program_id,
-            &merkle_trees,
-            &payer_pubkey,
-            &tree_nodes,
-            &tdas,
-            &claimants,
-            &claim_statuses,
-            micro_lamports_per_compute_unit,
-        )?;
+            micro_lamports,
+            keypair.pubkey(),
+        )
+        .await?;
+
         datapoint_info!(
-            "claim_mev_workflow-prepare_transactions",
-            ("epoch", merkle_trees.epoch, i64),
-            ("attempt", retries, i64),
-            ("tree_node_count", tree_nodes.len(), i64),
-            ("tda_count", tdas.len(), i64),
-            ("claimant_count", claimants.len(), i64),
-            ("claim_status_count", claim_statuses.len(), i64),
-            ("skipped_merkle_root_count", skipped_merkle_root_count, i64),
-            ("zero_lamports_count", zero_lamports_count, i64),
-            ("already_claimed_count", already_claimed_count, i64),
-            ("below_min_rent_count", below_min_rent_count, i64),
-            ("transaction_count", transactions.len(), i64),
-            (
-                "account_fetch_latency_us",
-                account_fetch_elapsed.as_micros(),
-                i64
-            ),
-            (
-                "transaction_prepare_latency_us",
-                start.elapsed().as_micros(),
-                i64
-            ),
+            "claim_mev_tips-send_summary",
+            ("claim_transactions_left", all_claim_transactions.len(), i64),
         );
 
-        if transactions.is_empty() {
-            info!("Finished claiming tips after {retries} retries, {failed_transaction_count} failed requests.");
-            return Ok(());
-        }
-
         if let Some((start_balance, desired_balance, sol_to_deposit)) = is_sufficient_balance(
-            &payer_pubkey,
-            &blockhash_rpc_client,
-            transactions.len() as u64,
+            &keypair.pubkey(),
+            &rpc_client,
+            all_claim_transactions.len() as u64,
```

Comment:
should we always pass in the full size? we only send 10k at a time.

also can move this up before `get_claim_transactions_for_valid_unclaimed()`, no need to get claim txns if we dont have enough money